nav_tui.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python
+import rospy
+# Curses is used for creating a user interface within a terminal
+import curses
+from curses import panel
+import numpy as np
+from std_msgs.msg import Float32, String  # Used for battery voltage and wrench mode, respectively
+from ros_alarms import AlarmListener  # Used for listening for kill alarm
+from geometry_msgs.msg import PointStamped  # Used for LLA gps coordinates
+from roboteq_msgs.msg import Status  # Used for Thruster faults
+
+
+class nav_tui(object):
+    '''
+    Text User Interface for diagnostics
+    '''
+
+    def __init__(self, screen):
+
+        '''
+        Initialization of Tui window and fields that will be assigned by callback functions
+        '''
+
+        self.y, self.x = screen.getmaxyx()
+        self.window = screen.subwin(self.y, self.x, 0, 0)
+        self.window.keypad(1)  # Allows for escape sequences (ctrl-c) to be interpreted by the curses window
+        self.panel = panel.new_panel(self.window)  # Creates a panel of window object, self.window
+        self.panel.hide()
+        panel.update_panels()
+        # The following are default field initalizations for various values that that will be overwritten
+        self.voltage = 0
+        self.wrench = None
+        self.LAT = None
+        self.LON = None
+        self.ALT = None
+        # The fault values are initalized as zero (no faults)
+        self.FL_fault = 0
+        self.FR_fault = 0
+        self.BL_fault = 0
+        self.BR_fault = 0
+        self.motor_topics = ['/FL_motor', '/FR_motor', '/BL_motor', '/BR_motor']
+        # The meaning of different fault codes
+        self.fault_codes = {1: 'OVERHEAT', 2: 'OVERVOLTAGE', 4: 'UNDERVOLTAGE', 8: 'SHORT_CIRCUIT',
+                            16: 'EMERGENCY_STOP', 32: 'SEPEX_EXCITATION_FAULT', 64: 'MOSFET_FAILURE',
+                            128: 'STARTUP_CONFIG_FAULT'}
+
+        # Will try and access low/critical voltage ros parameters, if none are located default values are used
+        self.lowThreshold = rospy.get_param('battery-voltage/low', default=26)
+        self.criticalThreshold = rospy.get_param('battery-voltage/critical', default=20)
+
+    # Cb function for wrench type
+    def wrench_cb(self, string):
+        self.wrench = string.data
```

Comment:
Helpful to trim the `/wrench` from this topic if it exists, so:
`/wrench/autonomous` -> `autonomous`
`autonomous` -> `autonomous`

Reply:
I added an if-condition for this. If it returns true that the string message contains: "/wrench/" and trims that part off, otherwise it just returns the string.